Floofbot/Modules/Filter.cs

```diff
@@ -0,0 +1,236 @@
+            }
+            return;
+        }
+        [Summary("Adds a new filtered word")]
+        [Command("add")]
+        public async Task AddFilteredWord([Summary("filtered word")]string word)
+        {
+            CheckServerEntryExists(Context.Guild.Id);
+            string newWord = word.ToLower();
+            bool wordAlreadyExists = CheckWordEntryExists(newWord, Context.Guild);
+
+            if (wordAlreadyExists)
+            {
+                await Context.Channel.SendMessageAsync($"{word} is already being filtered!");
+                return;
+            }
+            else
+            {
+                try
+                {
+                    _floofDb.Add(new FilteredWord
+                    {
+                        Word = newWord,
+                        ServerId = Context.Guild.Id
+                    });
+                    _floofDb.SaveChanges();
+                    await Context.Channel.SendMessageAsync($"{word} is now being filtered!");
+                }
+                catch (Exception ex)
+                {
+                    await Context.Channel.SendMessageAsync("An error occured: " + ex.Message);
+                    Serilog.Log.Error("Error when trying to add a filtered word: " + ex);
+                }
+            }
+        }
+        [Summary("Removed an existing filtered word")]
+        [Command("remove")]
+        public async Task RemoveFilteredWord([Summary("filtered word")]string word)
+        {
+            CheckServerEntryExists(Context.Guild.Id);
+            string oldWord = word.ToLower();
+            bool wordAlreadyExists = CheckWordEntryExists(oldWord, Context.Guild);
+
+            if (!wordAlreadyExists)
+            {
+                await Context.Channel.SendMessageAsync($"{word} isn't being filtered!");
+                return;
+            }
+            else
+            {
+                try
+                {
+                    FilteredWord wordEntry = _floofDb.FilteredWords.AsQueryable().Where(w => w.ServerId == Context.Guild.Id).Where(w => w.Word == oldWord).First();
+                    _floofDb.Remove(wordEntry);
+                    _floofDb.SaveChanges();
+                    await Context.Channel.SendMessageAsync($"{word} is no longer being filtered!");
+                }
+                catch (Exception ex)
+                {
+                    await Context.Channel.SendMessageAsync("An error occured: " + ex.Message);
+                    Serilog.Log.Error("Error when trying to toggle the channel filtering: " + ex);
+                }
+            }
+        }
+        [Summary("Lists all filtered words")]
+        [Command("list")]
+        public async Task ListFilteredWords()
+        {
+            List<FilteredWord> filteredWords = _floofDb.FilteredWords.AsQueryable()
+                .Where(x => x.ServerId == Context.Guild.Id)
+                .OrderBy(x => x.Id)
```

Comment:
Sanity check: we're ordering by time of insertion into the DB rather than alphabetical order here?

Reply:
yes, i believe thats how the current filter works too